### ISSUE TITLE
Configuring an external monitor and laptop monitor

### ISSUE THREAD

**Issue by hutch120** [MrMEEE/ironhide#66]
Hi All,

Thanks to many posts on this forum I've managed to get ironhide working on my Dell XPS L702x laptop. I feel I'm really close to getting this configured 100% but need a little assistance. In particular I'd note this comment as being most helpful.
  https://github.com/Mrmeee/ironhide/issues/2#issuecomment-1876688

I've scoured this forum and the xorg web site, and am coming up blank on getting an external monitor going while at the same time not killing my laptop monitor config. I'm not even concerned about having them running at the same time. Note that I am able to configure an external monitor by doing a manual mash up of  ironhide-settings/nvidia-setting, but then when I pull out the monitor cable and reboot I get a black screen and have to delete the xorg.conf file to recover.

I feel there must be a bunch of people on this forum that have solved this issue. I'd be greatful if they share their experience here.

I've spent quite some time trying to get this going and if I can get it I'll do a full write up. I've started to do a write up here:
  http://www.clearlysecure.com.au/node/90

Basically I think it is an issue with configuring xorg.conf.nvidia or maybe xorg.conf??

I'll attach the bug report, but note that ironhide is running 100%, I'm pretty sure this is just an xorg.conf configuration issue.

**Comment by jaimefma** [comment]
I am in the same situation with the same laptop. 

Can anybody configure the two monitors?

**Comment by MrMEEE** [comment]
I'm not having this problem on my Alienware, so i'm having a hard time testing.. But i'll look into it, and see what I can do.. maybe integrate something into ironhide-settings..

**Comment by hutch120** [comment]
Not much to report but had a play around with this some more yesterday and found the following. The nvidia setting GUI finds my Dell 2711 monitor when I have the HDMI cable plugged in, but when I save the config to xorg.conf and restart nothing comes up and I have to delete the xorg.conf. Also another strange thing is that the nvidia settings gui doesn't seem to detect my laptop monitor when the HDMI cable is plugged in meaning that it seems to save only the external monitor config, don't know if this helps.

On my old laptop, this is an image of my working NVIDIA X Server Settings on Ubuntu 10.04 Dell XPS M1530 Nvidia driver version195.36.24. It detects both my laptop monitor and external monitor.
http://tinypic.com/view.php?pic=167tisi&s=7

**Comment by hutch120** [comment]
I think I've got it! I've been so caught up with the nvidia settings and xorg.conf that I didn't think to just use the built in Ubuntu display settings tool, but it seems to work pretty well, and it even supports multi-monitor! Load it at System->Preferences->Monitors. With my "Display Port" cable plugged in then it  just detected my external Dell monitor and laptop monitor and I was able to set them up with individual resolutions, etc. Oh man, the number of hours I've spent on this and of course it is actually so easy.

So just to summarize, my configuration is no xorg.conf, the default xorg.conf.nvidia that ironhide generates during config. Then (with the display port cable, doesn't seem to play nice with the HDMI cable) use the built in Ubuntu Monitors config.

I set it up to have the external monitor on and the laptop monitor off, but it will load the laptop monitor if the cable is not plugged in as I would have expected. 

Thanks guys for your help, I think a dummies guide would be useful, hope these notes help someone else, if this was actually the way it was supposed to be configured. :P

Still not sure how all this works, and if I'm even using the graphics card to it's full potential, but at least I can use the laptop productively now.

**Comment by jaimefma** [comment]
@hutch120,

That doesn't work for me. 
1. When I start the nvidia-settings it says:

<i>"You do not appear to be using the NVIDIA X driver.  Please edit your X configuration file (just run `nvidia-xconfig` as root), and restart the X server."</i>

I supposed it is because the Ironhide. I don't have an xorg.conf file. 

I have tried the Ubuntu application to connect the external monitor. The gnome-display-properties application doesn't detect the monitor plugged in. 
1. I only have an HDMI port, so I cannot connect with anything else.

**Comment by hutch120** [comment]
Hi @Jaimefma,

Yeah, seems you can't use nvidia-settings directly, you need to run ironhide-settings and then pick the nvidia option (second last one), but that didn't help me in the end as I ended up running with no xorg.conf and using the built in Monitors tool.

I've put some more details about the testing I did on my blog (link below), maybe that will save you some testing time, but if you don't have a display port I'm not sure what the answer is because I only managed to get it going with the Display Port on my L702x.

http://www.clearlysecure.com.au/node/90

Hope you get it going, if you do I'd be interested in how.

Cheers,
Simon

**Comment by jaimefma** [comment]
Hello,

As you said before, with the HDMI cable it doesn't work. If I lunch the nvidia settings through ironhide, it detects the external monitor but not my laptop one. I don't know what to do. 

It's like each monitor work with different graphic card .... ¿Can anyone reopen this issue?  or I should open a different one?

**Comment by jaimefma** [comment]
Sorry, I forgot to say that I don't have any other port to plug-in the monitor. I have the nvidia GT 550M

**Comment by hutch120** [comment]
Please don't take this the wrong way, but are you sure? Looks like the entire Dell XPS range have display port now.
http://content.dell.com/au/en/home/d/help-me-choose/hmc-nvidia-graphics-xps-lt.aspx#hmcTab5
http://content.dell.com/us/en/home/d/help-me-choose/hmc-nvidia-graphics-xps-lt.aspx?ref=CFG#hmcTab5

You know Dell uses the Mini Display Port? It looks a lot like a large USB port (not the standard Display Port)?
http://en.wikipedia.org/wiki/Mini_DisplayPort

**Comment by graydw57** [comment]
(I have the same laptop.) First, let me state that I haven't even tried to get external monitors running (yet).

However, my understanding is the HDMI connector is wired directly to the Nvidia card. The displayport connector goes to
the on-board (Intel) graphics. Note that the Nvidia card _never_ talks to the display directly - that's why you have to choose 
the transport you want to use on Ironhide, when you do the configuration. Does anyone know how they connect up under
WIndows - just seems too clunky a mechanism for an OEM to ship. In other words, in a Windows environment, is there
some kind of switch, or what?

**Comment by hanynowsky** [comment]
In windows 7, the Nvidia-settings allows you to select the monitors you want to activate.

in the case of the Dell XPS 15:
- mini DisplayPort is linked to the onboard Intel HD 3000.
- HDMI is linked to the Nvidia card.

That is what Optimus is all about. Through Nvidia Configuration panel in Windows, you select the main monitors and Optimus just transmits the signal to the onboard graphics card. (Nvidia card cannot directly send signal to monitors, it must go through the onboard graphic cards).

As for now, bumblebee / ironhide manages the nvidia-settings, it should in the nearest future enable the access to the external monitor through HDMI.

**Comment by hanynowsky** [comment]
EDIT:

Ubuntu 11.10 and Bumblebee installed on a Dell XPS L502X :
- mini DisplayPort connected to a 23' 1920X1080 external monitor (works fine). (Enabled using Gnome Displays Settings)
- HDMI connected to another 23' 1920X1080 external monitor (not enabled and nothing output).
- Laptop screen disabled using Gnome Displays Settings
- Open terminal and do: $ optirun nvidia-settings -c :8

This will open Nvidia settings

Then do again on terminal:

DISPLAY=:8.0 gedit   or  DISPLAY=:8.0 firefox  or DISPLAY=:8.0 optirun gedit  

This will activate the monitor connected through HDMI and display gedit (text editor) with a resolution of 640X480 (The resolution cannot be changed in the nvidia-settings).

That is the best we reach untill now. Optimistic.... :)

**Comment by mauriz** [comment]
Hi All, 

I have a XPS L502X. I have an issue with the external monitor: using the Ubuntu utility I can set the resolution up to 1280x1024 although the monitor is able to get a resolution of 1920xsomething_I_don't_remember.

I successfully installed bumblebee (no  hironide): currently I don't have a xorg.conf file.
I am interested in having the external monitor working alone, i.e. turning off the screen of the laptop.

If the monitor on the display port depends on the intel card and I  can not see a xorg.conf file
, which kind of file should I look for?

Thanks

**Comment by hanynowsky** [comment]
Actually, what version of Ubuntu you installed? 11.04 or 11.10?  Kubuntu?
In my case Ubuntu 11.10 recognizes the external monitor's native resolution out of the box. No need for a xorg file.
type the command :  xrandr  in your terminal to see if the native resolution is indeed recognized or not.
What cable do you use to connect your external monitor to the mini DisplayPort (some bad quality cables might not able to support high resolutions).
In my case, I use a mini DisplayPort to DVI cable (10 bucks off ebay) which supports a maximum of 1920X1080.

Turning off the  laptop screen and working with the external monitor can be achieved through the Gnome Display Settings.
You have a button to turn off the internal screen. And menu box to select the external monitor's resolution.
Turn off the Internal screen first, then go to the external monitor and see if you have :  1920X1080 there.

**Comment by mauriz** [comment]
I have Ubuntu 11.10 and a  mini DisplayPort to DVI cable  (on windows I can run full resolution). Using gnome display I can achieve a max resolution of 1280x1024.

Currently I am not in my office and I can not plug the external monitor, I will try with xrandr and possibly add another item with 1920x1440.
Does it seem the right procedure?

**Comment by hanynowsky** [comment]
Through XRANDR you can of course achieve it.

**Comment by mauriz** [comment]
Hi,

I still don't get it. I tried the following:

cvt 1920 1080

that outputs:

1920x1080 59.96 Hz (CVT 2.07M9) hsync: 67.16 kHz; pclk: 173.00 MHz

then 

xrandr --newmode "1920x1080_60.00"  173.00  1920 2048 2248 2576  1080 1083 1088 1120 -hsync +vsync
xrandr --addmode DP1 1920x1080_60.00

But when I select this configuration from the gnome display it replies:

"could not set the configuration for CRTC 63"

The output of my xrandr command currently is:

Screen 0: minimum 320 x 200, current 1920 x 1080, maximum 8192 x 8192
LVDS1 connected (normal left inverted right x axis y axis)
   1366x768       60.1 +
   1360x768       59.8     60.0  
   1024x768       60.0  
   800x600        60.3     56.2  
   640x480        59.9  
VGA2 disconnected (normal left inverted right x axis y axis)
HDMI2 disconnected (normal left inverted right x axis y axis)
DP1 connected 1280x1024+0+0 (normal left inverted right x axis y axis) 531mm x 299mm
   1280x1024      75.0\*    60.0  
   1152x864       75.0  
   1024x768       75.1     60.0  
   800x600        75.0     60.3  
   640x480        75.0     60.0  
   720x400        70.1  
   1920x1080_60.00   60.0

**Comment by mauriz** [comment]
Hi, 

I got a lower resolution at 1600x900.
But I am not able to get 1920x1080, the one that I get on windows.

Any ideas?

**Comment by hanynowsky** [comment]
I still do not get it neither! Seems fine to me! There is some kind of conflict there! 
Maybe the information you entered through XRANDR is not correct. The native resolution should be on top like mine:
(Note also that in Dell XPS L502X, DP1 and HDMI1 are inverted):

hanine@L502X:~$ xrandr -q
Screen 0: minimum 320 x 200, current 1920 x 1080, maximum 8192 x 8192
LVDS1 connected (normal left inverted right x axis y axis)
   1366x768       60.0 +
   1360x768       59.8     60.0  
   1024x768       60.0  
   800x600        60.3     56.2  
   640x480        59.9  
VGA1 disconnected (normal left inverted right x axis y axis)
HDMI1 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 510mm x 287mm
   1920x1080      60.0*+
   1600x1200      60.0  
   1680x1050      60.0  
   1280x1024      60.0  
   1440x900       59.9  
   1280x960       60.0  
   1280x800       59.8  
   1024x768       60.0  
   800x600        60.3     56.2  
   640x480        60.0  
DP1 disconnected (normal left inverted right x axis y axis)

PS: my external monitor is a Samsung SynMaster (1920X1080p) connected through mini DisplayPort - mini DP to DVI -

**Comment by mauriz** [comment]
but are you using the HDMI port?

**Comment by hanynowsky** [comment]
No I am using the Mini DisplayPOrt. But XRANDR believes its HDMI.

**Comment by mauriz** [comment]
I don't have this issue: as you can see above, my PD1 results connected. However, I can not set above 1600x900...do you think something wrong happens with gathering  data using cvt?

**Comment by hanynowsky** [comment]
No! It is fine for me : my cvt 1920 1080 is :   

# 1920x1080 59.96 Hz (CVT 2.07M9) hsync: 67.16 kHz; pclk: 173.00 MHz

Modeline "1920x1080_60.00"  173.00  1920 2048 2248 2576  1080 1083 1088 1120 -hsync +vsync

But I believe the error's origin is your command :

xrandr --newmode "1920x1080_60.00" 173.00 1920 2048 2248 2576 1080 1083 1088 1120 -hsync +vsync
xrandr --addmode DP1 1920x1080_60.00

The mode 1920x1080_60.00  is not recognized as a mode! and the refresh rate is likely to be 59.96 not 60

Classically speaking here are the steps as an example:

Here are the steps
Use xrandr to make sure that the new mode can fit within the maximum framebuffer size
Code:
xrandr | grep maximum
Use gtf to create a mode line
Code:
gtf 1440 900 59.9
Add new mode using xrandr
Code:
xrandr --newmode "1440x900_59.90"  106.29  1440 1520 1672 1904  900 901 904 932  -HSync +Vsync
Add this newly added mode to the desired output (VGA/LVDS etc)
Code:
xrandr --addmode VGA 1440x900_59.90
Choose the new mode
Code:
xrandr --output VGA --mode 1440x900_59.90

**Comment by mauriz** [comment]
Sorry, 

but I did not understand: which is the error? I also run:

xrandr --newmode "1920x1080" .......
xrandr --addmode DP1 1920x1080

These commands do not give me any warning about possible not recognized modes.

I also did not understand the frequency: why did you put 59.9Hz? Is it particular of your screen or in general 59.9 instead of 60 is needed?

Thanks

**Comment by hanynowsky** [comment]
Please try with 59.9 instead of 60 and see!!!
I myself am not sure why the clock of 59Hz must be explicitly forced instead of 60Hz (Most LCD run on 59 or 59.9 Hz)!
DId you have the same issue before installing ironhide?
One thing to test: Boot with the 11.10 Live CD or USB Key and see if the native resolution is detected.
I think we're going to be killed as this issue is not related to IRONHIDE :)

**Comment by mauriz** [comment]
Tried, but it doesn't work. 

When I try 

gtf 1920 1080 59.9

I get

  # 1920x1080 @ 59.90 Hz (GTF) hsync: 66.97 kHz; pclk: 172.51 MHz
  Modeline "1920x1080_59.90"  172.51  1920 2040 2248 2576  1080 1081 1084 1118  -HSync +Vsync

But then it is not possible to apply this mode. It repeats

could not set the configuration for CRTC 63

**Comment by rawsp33d** [comment]
This issue is marked as closed, but the problem still exists a year later.

**Comment by ryandpardey** [comment]
And now a year later, the problem still exists. See my thread here:

http://ubuntuforums.org/showthread.php?t=2161010
